mode2.lua

```diff
@@ -961,43 +978,63 @@ function courseplay:unload_combine(vehicle, dt)
 		if distance_to_wp > vehicle.cp.shortestDistToWp and distance_to_wp < 3 then
 			distToChange = distance_to_wp + 1
 		end
-		if distance_to_wp < distToChange then
-			--[[if vehicle.cp.mode2nextState == 81 then
-				if vehicle.cp.activeCombine ~= nil then
-					courseplay:unregisterFromCombine(vehicle, vehicle.cp.activeCombine)
-				end
-			end]]
 
+		if distance_to_wp < distToChange then
+      -- Switching to next waypoint
 			vehicle.cp.shortestDistToWp = nil
 			if #(vehicle.cp.nextTargets) > 0 then
-				-- courseplay:setModeState(vehicle, 5);
-				courseplay:setCurrentTargetFromList(vehicle, 1);
+        -- still have waypoints left
+        -- if we are chasing our combine, check if we are close enough now (this can happen when
```

Comment:
Still not sure if this is needed or is already being taken care of some code elsewhere?